tools/do_meili_tools.py

```diff
@@ -1,23 +1,53 @@
 import time
 import socket
+import requests
 
 def wait_for_droplet_creation(droplet):
     while True:
         d = droplet.get_actions()
         if d[0].type == "create" and d[0].status == "completed":
-            droplet = droplet.load()
-            print("IP:", droplet.ip_address, "id:", droplet.id)
             break
         time.sleep(2)
 
 def wait_for_ssh_availability(droplet):
     while True:
+        time.sleep(2)
         s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
             s.connect((droplet.ip_address, 22))
             s.shutdown(2)
             s.close()
-            break
+            return
         except Exception:
             continue
-        time.sleep(2)
\ No newline at end of file
+
+def wait_for_health_check(droplet):
+    while True:
+        time.sleep(2)
+        try:
+            resp = requests.get("http://{}/health".format(droplet.ip_address))
+            if resp.status_code >=200 and resp.status_code < 300:
+                return
+        except Exception as e:
+                continue
+
+def wait_for_droplet_shutdown(droplet):
+    while True:
+        try:
+            d = droplet.get_actions()
+            for act in d:
+                if act.type == "shutdown" and d[0].status == "completed":
+                    return
+        except Exception as e:
+            print("   Exception: {}".format(e))
+            return
+
+def wait_for_snapshot_creation(droplet):
+    while True:
+        try:
+            time.sleep(2)
+            d = droplet.get_actions()
+            if d[0].type == "snapshot" and d[0].status == "completed":
+                return
+        except Exception as e:
+            continue
```

Comment:
If there is an error, shouldn't it better to `print()` and `return` instead of `continue`?

Reply:
I was thinking. This is asking the DO API if the task is completed. An error here doesn't mean an error in the task or the script, but an error communicating with this API. Maybe the safes this is to set a sleep(300) to leave it the time to finish the snapshot. Because if we return here, next instruction will be "Destroy Droplet", and I certainly don't want to destroy it because of this error. WDYT?

Reply:
I was confused, I thought it would be an error sent by MeiliSearch.
It makes sense to continue then.

Reply:
But not to continue right away. The task will still needs time, even if this function fails. Which is why I would say "if I can't know when DO finishes the droplet, I'll just wait a few minutes, long enough for it to make a snapshot safely" => sleep()

Reply:
Seems good!